components/work_templates/components/menu.tsx

```diff
@@ -70,21 +69,23 @@ const Menu = (props: MenuProps) => {
     const worktemplates = useSelector((state: GlobalState) => state.entities.worktemplates.templatesInCategory);
 
     useEffect(() => {
-        // DEV - REMOVE ME WHEN SERVER RETURNS DATA
-        // allows us to load the dev work templates from the store
-        dispatch(injectDevData(workTemplateDevData));
+        if (categories?.length) {
+            return;
+        }
+        dispatch(getWorkTemplateCategories());
     }, []);
 
     useEffect(() => {
-        if (categories?.length === 0 || selectedCategory !== '') {
+        if (categories?.length === 0) {
             return;
         }
-
         setSelectedCategory(categories[0].id);
-    }, [categories, selectedCategory]);
+        dispatch(getWorkTemplates(categories[0].id));
```

Comment:
first and very important question, does the categories and worktemplates need to exist in the redux global state? I ask because if there is no need to do it, it will simplify a lot your code by handling this information in your local component state and obviously will remove this extra info from global memory.
second, we should find a way to memoize the worktemplates data so it doesn't dispatch the api request every time the user changes the category.

Reply:
Good point @pvev. I started with the understanding that we do need global state but I am not sure where and how it would be useful. @JulienTant, whats your take on this ?

Reply:
I don't have a strong opinion on one vs the other but we don't need it in the global state. 
I wonder if those actions should not be placed in `work_templates/index.tsx` because we will need them when sending the "execute this template" command and it's not gonna be in the menu.

Something like

```tsx
<Menu categories={listOfCategories} templatesForCategory={loadedTemplate} onCategoryChange={onCategoryChange} .../>
```

Would that make sense? The advantage of the store is that those can be requested from anywhere, but if we want to be conservative with our store usage we'll have to pass things down to subcomponents when needed and trigger handlers when things need to be updated higher up (we do have some of this already)

Reply:
Thanks @JulienTant. I have a feeling that lets continue with the redux state. While, we do not intend to use the worktemplates or categories anywhere outside the context of Work Templates but still the redux state would be useful as with it we could avoid props drilling and callback functions call over multiple files. With redux store, I feel, it would lead to less confusion and more clarity .

Reply:
> second, we should find a way to memoize the worktemplates data so it doesn't dispatch the api request every time the user changes the category.

If the data lives in the store, we can check in the store if we have work templates for this category.

- If we do then don't need for a request
- If we don't then do the request

In any case, the first load of the modal must retrieve fresh data  to make sure we don't get stuck with old data. Somehow I'm back on thinking we should move those call on the index.tsx file haha.

Reply:
ok, thanks for the responses guys!